src/axom/sidre/core/SidreDataTypeIds.h

```diff
@@ -19,7 +19,7 @@
 
 #include <stdint.h> /* for int64_t */
 
-using SIDRE_IndexType = int64_t;
+typedef int64_t SIDRE_IndexType;
```

Comment:
Minor: would it make sense to add a (C-style) comment to the top of this file that it needs to be compatible with `C` ?

Reply:
My assumption was that the `.c` suffix was sufficient.  But given that most of AXOM is C++ it wouldn't hurt to emphasis the point.

Reply:
Here's what we have in `config.hpp.in`:

https://github.com/LLNL/axom/blob/2ad90f2e4ec001241f41102992575bf07a6bee30/src/axom/config.hpp.in#L11-L14

This one would probably need to say something about the file containing "valid `C`" as opposed to just "`C`-style comments"

Reply:
Should that file be renamed `config.h.in`? (In another PR)

Reply:
Maybe (?) 

But it would require all of our users to update their axom usage, so it might not make sense unless there was a good technical reason, e.g. someone was trying to use axom with only a C compiler and the code wouldn't compile...

If we do make the change, perhaps we should line it up with a release ?

Reply:
I added the comment.  Now it's up to people to read it before changing the file.